spyder/plugins/editor/widgets/tests/test_editor_and_outline.py

```diff
@@ -230,7 +258,158 @@ def test_toggle_on_show_all_files(editorstack, outlineexplorer, test_files):
+    def get_tree_elements(treewidget):
+        root_item = treewidget.get_top_level_items()[0]
+        root_ref = root_item.ref
+        filename = osp.basename(root_ref.name)
+        root_tree = {filename: []}
+        stack = [(root_tree[filename], node) for node in root_ref.children]
+
+        while len(stack) > 0:
+            parent_tree, node = stack.pop(0)
+            this_tree = {node.name: []}
+            parent_tree.append(this_tree)
+            this_stack = [(this_tree[node.name], child)
+                          for child in node.children]
+            stack = this_stack + stack
+        return root_tree
+
+    code_editor, outlineexplorer = lsp_codeeditor_outline
+    treewidget = outlineexplorer.treewidget
+
+    case_info = CASES['text']
+    filename = case_info['file']
+    tree_file = case_info['tree']
+
+    with open(filename, 'r') as f:
+        lines = f.read()
+
+    with open(tree_file, 'r') as f:
+        trees = json.load(f)
+
+    code_editor.toggle_automatic_completions(False)
+    code_editor.toggle_code_snippets(False)
+
+    # Set cursor to start
+    code_editor.set_text('')
+    code_editor.go_to_line(1)
+
+    # Put example text in editor
+    code_editor.set_text(lines)
+    with qtbot.waitSignal(code_editor.lsp_response_signal, timeout=30000):
+        code_editor.document_did_change()
+
+    with qtbot.waitSignal(code_editor.lsp_response_signal, timeout=30000):
+        code_editor.request_symbols()
+
+    # Check that the outline tree was initialized successfully
+    tree = trees[0]
+    root_tree = get_tree_elements(treewidget)
+    assert root_tree == tree
+
+    # Remove "d" symbol
+    code_editor.go_to_line(14)
+    cursor = code_editor.textCursor()
+    start = code_editor.get_position_line_number(13, -1)
+    end = code_editor.get_position_line_number(17, 0)
+    cursor.setPosition(start)
+    cursor.setPosition(end, QTextCursor.KeepAnchor)
+    code_editor.setTextCursor(cursor)
+    code_editor.cut()
+
+    with qtbot.waitSignal(code_editor.lsp_response_signal, timeout=30000):
+        code_editor.document_did_change()
+
+    with qtbot.waitSignal(treewidget.sig_tree_updated, timeout=30000):
+        code_editor.request_symbols()
+
+    tree = trees[1]
+    root_tree = get_tree_elements(treewidget)
+    assert root_tree == tree
+
+    # Add "d" symbol elsewhere
+    code_editor.go_to_line(36)
+
+    with qtbot.waitSignal(code_editor.lsp_response_signal, timeout=30000):
+        qtbot.keyPress(code_editor, Qt.Key_Return)
+        qtbot.keyPress(code_editor, Qt.Key_Return)
+
+    qtbot.keyPress(code_editor, Qt.Key_Up)
+    code_editor.paste()
+
+    with qtbot.waitSignal(code_editor.lsp_response_signal, timeout=30000):
+        code_editor.document_did_change()
+
+    with qtbot.waitSignal(treewidget.sig_tree_updated, timeout=30000):
+        code_editor.request_symbols()
+
+    tree = trees[2]
+    root_tree = get_tree_elements(treewidget)
+    assert root_tree == tree
+
+    # Move method1
```

Comment:
I think `Move` should be `Remove` here?

Reply:
It is move because of the cut operation performed below